src/components/ListItem.jsx

```diff
@@ -16,29 +19,25 @@ export function ListItem({ item }) {
 	useEffect(() => {
 		if (dateLastPurchased) {
 			const is24HoursPassed = has24HoursPassed(dateLastPurchased);
-			if (is24HoursPassed) {
-				setChecked(false);
-			} else {
-				setChecked(true);
-			}
+			setChecked(!is24HoursPassed);
 		}
 	}, [dateLastPurchased]);
 
 	const handleChange = () => {
 		if (!checked) {
 			setChecked(!checked);
 			const listPath = localStorage.getItem('tcl-shopping-list-path');
-			updateItem(listPath, {
-				itemName: name,
-				dateLastPurchased: checked ? new Date() : null,
-				totalPurchases: increment(1),
+
+			updateItem(listPath, id, {
+				...item,
 			})
```

Comment:
<!-- Nitpicks are trivial preference-based requests.  -->
**nitpick ( non-blocking ):** You can just pass in the item here now:
```suggestion
			updateItem(listPath, id, item)
```